src/main/java/seedu/address/logic/commands/DeleteTagCommand.java

```diff
@@ -38,11 +41,37 @@ public CommandResult execute(Model model) throws CommandException {
         if (!isSuccessful) {
             throw new CommandException(MESSAGE_NONEXISTENT);
         }
+
+        for (Tag tag : tags) {
+            removeTagFromPersons(model, tag);
+        }
+
         String successMessage = MESSAGE_SUCCESS + tags + "\n";
         String currentTags = YOUR_TAGS_PREFIX + model.getTagList();
         return new CommandResult(successMessage + currentTags);
     }
 
+    /**
+     * Removes the deleted {@code Tag} from all persons in the address book.
+     */
+    private void removeTagFromPersons(Model model, Tag tag) {
+        List<Person> persons = model.getFullPersonList();
+        for (Person person : persons) {
+            if (person.getTags().contains(tag)) {
```

Comment:
Could this potentially violate the Law of Demeter (don't talk to strangers)? Maybe this condition can be handled within the `Person` class with a method like `.tagsContain(Tag tag)`